src/scripts/main.js

```diff
@@ -1,3 +1,60 @@
 'use strict';
 
-// write code here
+const BASE_URL = 'https://mate-academy.github.io/phone-catalogue-static/api';
+const ENDPOINTS = {
+  phones: '/phones.json',
+  phoneById: (id) => `/phones/${id}.json`,
+};
+
+const createDOM = (className, title, ids, position) => {
+  document.querySelector('body').insertAdjacentHTML(position, `
+    <div class="${className}">
+      <h2>${title}</h2>
+      <ul>
+        ${ids.map(id => `<li>${id.name}</li>`).join('')}
+      </ul>
+    </div>
+  `);
+};
+
+const request = (url) => {
+  return fetch(`${BASE_URL}${url}`)
+    .then(value => value.json())
+    .catch(error => Promise.reject(error));
+};
+
+const getPhones = () => request(ENDPOINTS.phones);
+
+const map = (ids) => ids.map(id => request(ENDPOINTS.phoneById(id)));
```

Comment:
Maybe it's better call like mapIds or else, because you can confuse other people with the same method map

Reply:
agree ⬆️